pkg/events/event_publisher.go

```diff
@@ -0,0 +1,89 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package events
+
+import (
+	"sync"
+	"time"
+
+	"github.com/apache/incubator-yunikorn-core/pkg/log"
+	"github.com/apache/incubator-yunikorn-core/pkg/plugins"
+	"go.uber.org/zap"
+)
+
+type EventPublisher interface {
+	StartService()
+	Stop()
+	GetEventStore() EventStore
+}
+
+type shimPublisher struct {
+	store EventStore
+	stopped bool
+
+	sync.Mutex
+}
+
+func NewShimPublisher(event EventStore) EventPublisher {
+	return &shimPublisher{
+		store:   event,
+		stopped: false,
+	}
+}
+
+func (sp *shimPublisher) isStopped() bool {
+	sp.Lock()
+	defer sp.Unlock()
+
+	return sp.stopped
+}
+
+func (sp *shimPublisher) StartService() {
+	go func () {
+		for {
+			if sp.isStopped() {
+				break
+			}
+			if eventPlugin := plugins.GetEventPlugin(); eventPlugin != nil {
+				messages := sp.store.CollectEvents()
+				log.Logger().Debug("Sending events", zap.Int("number of messages", len(messages)))
```

Comment:
How do we handle requests from a certain point onwards ?
Do we always get all events or is there a limit (like 500 or possible) ?

Reply:
not at this point in time, we'll work on this when we see more perf numbers